pysolr.py

```diff
@@ -898,11 +905,12 @@ def add(self, docs, boost=None, fieldUpdates=None, commit=True, softCommit=False
         m = force_unicode(m)
 
         end_time = time.time()
+        commit = self._get_commit(commit)
```

Comment:
Do we really need to make this call here when it's just calling `_update()` which does the same check? I'm thinking we could remove it in both `_add` & `_delete` and inline the `_get_commit` check into `_update`.